src/transformers/cache_utils.py

```diff
@@ -1590,6 +1392,94 @@ def is_compileable(self) -> bool:
 ### Deprecated classes
 
 
+class OffloadedCache(DynamicCache):
+    def __init__(self) -> None:
+        logger.warning_once(
+            "`OffloadedCache` is deprecated and will be removed in version v4.58 "
```

Comment:
uhmmm since these classes have no maintenance requirements on our side, should we consider a longer deprecation cycle?

(after these classes are removed, small things like `from transformers import OffloadedCache` will break the user's code 😢 )